tokio/src/runtime/builder.rs

```diff
@@ -1522,11 +1522,16 @@ impl Builder {
     }
 
     #[cfg(tokio_unstable)]
-    fn build_current_thread_local_runtime(&mut self) -> io::Result<LocalRuntime> {
+    fn build_current_thread_local_runtime(
+        &mut self,
+        opts: LocalOptions,
+    ) -> io::Result<LocalRuntime> {
         use crate::runtime::local_runtime::LocalRuntimeScheduler;
 
         let tid = std::thread::current().id();
 
+        self.before_park = opts.before_park;
+        self.after_unpark = opts.after_unpark;
```

Comment:
Is adding the support for the following hooks also safe? If so, we should also add them.

* `on_after_task_poll`
* `on_before_task_poll`
* `on_task_spawn`
* `on_task_terminate`